src/ffi.rs

```diff
@@ -64,6 +70,38 @@ impl Drop for ada_owned_string {
     }
 }
 
+/// Represents an std::vector<std::string>
```

Comment:
i don't know if it's worth it in this case, but you _could_ use https://github.com/rust-lang/rust-bindgen

Reply:
This requires a lot of refactoring as far as I can tell. Ref: https://cpp-api.ada-url.com/dd/dd6/ada__c_8cpp

Reply:
Yeah I think cxx is a better approach but I don't have the energy to do a rewrite at this point.